lib/active_model/serializer.rb

```diff
@@ -90,15 +90,27 @@ def root_name
       end
 
       def attributes(*attrs)
+        # Use `class_eval` rather than `define_method` for faster access
+        # and avoid retaining objects in the closure.
+        # Batch all methods in a single `class_eval` for efficiceny,
+        # and define all methods on the same line so the eventual backtrace
+        # properly maps to the `attributes` call.
+        source = ["# frozen_string_literal: true\n"]
```

Comment:
@casperisfine Thanks for this.  I've never seen a class_eval with the magic comment header. On the one hand I'm surprised the context isn't inherited by the eval, on the other hand, it makes a certain amount of sense thought evaluated input would have a new context.